### PR TITLE
Update references to population sizes of human worlds

### PR DESCRIPTION
**Pull request by Amazinite**
**Content**

## TL;DR (Isn't that what a summary is suppose to be?)

I simulated population sizes of all the worlds in human space. Some planets have mentions of their population size, but the numbers I ended up with don't match the descriptions. Instead of lowering the total number of people as given by MZ at roughly 100 billion, and instead of bringing up the population density of lesser inhabited planets by an insane amount just so I could reach that final number without clashing with the planet descriptions that existed, I've decided to change some descriptions according to what I simulated, essentially canonizing the numbers I came up with.

Link to my numbers is [here](https://docs.google.com/spreadsheets/d/1dWKimAiiXrlVCibdjyiFNqtQJdiXJqB7F0Cs9La0Z60/edit?usp=sharing).

The sheet also includes settlement dates for planets where they could be easily known. No, I'm not going to finish creating settlement dates for all planets right now. Yes, I'll eventually do it later, and we can canonize that too.

## (The not so) Summary
A while back now, I asked MZ for population numbers on the various species in the game, and he gave some rough numbers [here](https://github.com/endless-sky/endless-sky/issues/2540#issuecomment-302986796). A question that that has begged though is how those roughly 100 billion humans are spread out across the galaxy. Only a few planet descriptions actually mention anything concrete about population size of the planet, those being Valhalla, Midgard, Thule, Sunracer, Clark, Heartland, New Sahara, Sinter, Prime, Zug, and Thunder. We then learn from the Shuttle intro how many people Earth has. So great! That's 12 planets out of... 157 accounted for. Some other planet descriptions give numbers on a subset of the population, like millions of Amish on Harmony and tens of thousands of homeless on Bourne, but without more information on the proportion of the population within those groups, the descriptions really only give us lower bounds.

There are some more things we can look to, like how Earth is more than twice as populous as the next closest planet, and Chiron is the second most populous planet, but for the most part we don't have much in the way of figuring out how populated various regions of space are. At best we get a description on many worlds of them being sparsely populated or only having a single settlement or only having towns and no cities or being overpopulated. Well what defines a world as sparsely populated? How many people can a single settlement hold? How many people need to be in a region before it's considered a city? How many people does your average town have? How many people does a planet have to be considered overpopulated? Is overpopulation a factor of resources per person or of number of people? We might look to how we'd define these terms on Earth today, but I imagine that cities of the future are far larger and more impressive than those today, so it takes more to be considered a major city in the Republic.

I've wanted for a while now to try and make out how many people are on each planet, but it's only recently after some discussion on Discord that I actually cared to get around to it. Over the course of three days I read through every planet description in human space, as well as looked at all the attributes on all the planets, in order to get a rough idea of the ranking of the planets. Some planets were easy, like I mentioned above, providing a rough number for me. Most though mentioned nothing of population and only described the state of the planet. Using those descriptions of how nice of a place the planet was, and in some cases giving me a timeline of when the planet was settled, I was able to roughly rank all the planets in terms of population by assigning them random population numbers. These original numbers I came up with only came out to about 60 billion, but using the rankings I was able to create a function that would give me population numbers for a planet depending on its rank (although I did have to fiddle with it by often skipping ranks, e.g. most populous planet 20 and most populous planet 21 might have actually been considered rank 20 and rank 22 in the function, and toward the end planet 140 was actually like rank 2000 in the function, but whatever the case I got a distribution I like and a total value that is roughly around the original number of 100 billion).

For the curious, the function I used was an altered curve that describes the current populations of countries on Earth, scaled to fit Earth at about 20 billion. Seeing as how a good number of planets are named `"New <country>"` or `"New <city>"`, comparing planets to countries doesn't seem too out there.

My final spreadsheet for the population of planets can be found [here](https://docs.google.com/spreadsheets/d/1dWKimAiiXrlVCibdjyiFNqtQJdiXJqB7F0Cs9La0Z60/edit?usp=sharing). The top of this spreadsheet includes the population of groups of humans, the first being humanity as a whole, then going down into regions of humanity, some of which overlap. The "no planet of residence" line is a random number that I pulled out that refers to the number of people who basically live on ships, so merchant captains and their crew, or militia members, or maybe even Navy and Syndicated Security members that spend most of their time amongst the stars. The sheet then sorts the planets into regions, sorting the regions alphabetically and sorting the planets within the regions by population size.

As you might be able to see though, I've changed the population sizes of some planets from how they were described. I already mentioned how my original numbers came out to about 60 billion. Well in order to get 40 billion more people, I would have had to create many, many planets with close to or over a billion people on them, and planets described as sparsely populated would have a lot more people on them than I would expect. So instead of trying to force 40 billion into the upper limit that we had, which was 10 billion on Earth and 5 billion at most on any other planet, I decided to tweak some of the numbers we had. This includes bumping Earth's population from 10 billion to 20 billion, which allowed me to drag some other planets up with that, such as increasing Chiron's population from about 5 billion to about 10 billion, and Moonshake's population from the 3.5 or 4 billion that I originally had to about 7 billion. All these changes allowed for me to reach a total population that was close to 100 billion.

While this PR only changes descriptions to match my spreadsheet, everything from my spreadsheet can't be found in game. This is where #3663 gets involved, which entails providing information like this to the player in-game in an in-universe way. My idea as to how this data might be provided in-universe is by saying that this is census data. Perhaps the Republic does a census every 20 or 25 years, and 3000 was when the last census was done. (You can also see that I simulated what the population growth would need to be to hit 100 billion people by 3013, game start, and the answer is an average of 0.195% per year for all of humanity.) The player could then probably get this census data from Alexandria. Other than that, perhaps we create spaceport news that only appears on an exact planet for every planet, giving the player more information on how life is on each planet in the game, including mentions of the population size (perhaps not exact, but rough numbers).

## Save File
N/A

## PR Checklist
N/A

### CODE REVIEW

**Comment by Bardagh** [comment]
Much respect and appreciation.
